File: order-service/src/main/java/com/intexsoft/webshop/orderservice/dto/detail/DetailCreateDto.java
package com.intexsoft.webshop.orderservice.dto.detail;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@JsonIgnoreProperties(ignoreUnknown = true)
public class DetailCreateDto {
    @NotNull
    @Positive
    Long productId;
    @NotNull
    @Positive
    BigDecimal productPrice;
    @NotNull
    @Positive
    Short quantity;
}

File: order-service-kt/src/main/kotlin/com/intexsoft/webshop/orderservicekt/dto/detail/DetailCreateDto.kt
package com.intexsoft.webshop.orderservicekt.dto.detail

import jakarta.validation.constraints.Positive
import java.math.BigDecimal

data class DetailCreateDto(
    @field:Positive
    val productId: Long,
    @field:Positive
    val productPrice: BigDecimal,
    @field:Positive
    val quantity: Short
)